lib/io/rfdhdtDockerUtil.ts

```diff
@@ -0,0 +1,92 @@
+import * as fs from 'fs';
+import * as Path from 'path';
+import type * as Docker from 'dockerode';
+
+const SUPPORTED_FORMAT = new Map([
+  [ '.nq', 'nquad' ],
+  [ '.nt', 'ntriples' ],
+  [ '.ttl', 'turtle' ],
+  [ '.xml', 'rdfxml' ],
+  [ '.n3', 'n3' ],
+]);
+
+const ERROR_STEAM_FILE = fs.createWriteStream('./error_log_docker_rfdhdt');
+
+const RDF_HDT_IMAGE_REPO = 'rfdhdt/hdt-cpp';
+
+/**
+ * Pull the rfdhdt/hdt-cpp with docker
+ * equivalent of the docker command
+ * docker pull rfdhdt/hdt-cpp
+ * @param {Docker} docker - docker instance
+ */
+export async function pullHdtCppDockerImage(docker: Docker): Promise<void> {
+  return new Promise(async(resolve, reject) => {
+    const stream = await docker.pull(RDF_HDT_IMAGE_REPO, {});
+
+    const onFinished = (err: any): undefined => {
+      if (err === undefined || err === null) {
+        resolve();
+      } else {
+        // eslint-disable-next-line no-console
+        console.info(err);
+        reject(err);
+      }
+    };
+
+    const onProgress = (event: any): undefined => {
+      // eslint-disable-next-line no-console
+      console.info(event);
+    };
+
+    docker.modem.followProgress(stream, onFinished, onProgress);
```

Comment:
Is this line really needed?
Won't `docker.pull` just reject if an error occurs?

Reply:
I thought that a user would like to see the progress and the specific error. But I can make it simpler.